core/src/test/scala/org/apache/spark/LocalSparkContext.scala

```diff
@@ -31,6 +31,8 @@ trait LocalSparkContext extends BeforeAndAfterEach with BeforeAndAfterAll { self
 
   override def beforeAll(): Unit = {
     super.beforeAll()
+    // Clear default profile, just in case default profile is not cleaned in other suites.
+    ResourceProfile.clearDefaultProfile()
```

Comment:
seems like this should be unnecessary, were you getting test issues with this?

Reply:
Yes, without this change, one of the test suites failed because of unexpected default profile.

Reply:
it might be nice to look at why and update that suite if its not cleaning up properly. if its something we can't get around then fine.

Reply:
Will take a look.

Reply:
Done. After cleaning up default resource profile in `KubernetesClusterSchedulerBackendSuite` and `KubernetesExecutorBuilderSuite`, all UTs passed. I removed the code here.